### PR TITLE
AI weapon state

### PR DESCRIPTION
**Pull request by rlcevg**
This is basically a stub from [lua](https://github.com/spring/spring/blob/d7ebbb065cf8dcb6218ac2614a9830634623a8d7/rts/Lua/LuaSyncedRead.cpp#L3233), except that ally check done only once in skirmishAiCallback_Unit_getWeapon(s) while creating Weapon objects.

[Related discussion.](https://springrts.com/phpbb/viewtopic.php?f=15&t=33559)

Weapon usage example:
```C
Unit* u = ...;
auto wps = std::move(u->GetWeapons());
for (Weapon* wp : wps) {
  if (wp->GetReloadFrame() <= currentFrame) printf("Ready to fire\n");
  delete wp;
}
...
WeaponMount* wpMnt = ...;
Weapon* wp = u->GetWeapon(wpMnt);
if (wp->GetReloadFrame() <= currentFrame) printf("Ready to fire\n");
delete wp;
```

### CODE REVIEW

File: rts/ExternalAI/SSkirmishAICallbackImpl.cpp [hunk 1]
```diff
@@ -3842,6 +3843,26 @@ EXPORT(int) skirmishAiCallback_Unit_getLastUserOrderFrame(int skirmishAIId, int
...
 
+EXPORT(int) skirmishAiCallback_Unit_getWeapons(int skirmishAIId, int unitId) {
+	const CUnit* unit = getUnit(unitId);
+	if (unit && (isAlliedUnit(skirmishAIId, unit) || skirmishAiCallback_Cheats_isEnabled(skirmishAIId))) {
+		return unit->weapons.size();
+	}
+	return 0;
+}
+
+EXPORT(int) skirmishAiCallback_Unit_getWeapon(int skirmishAIId, int unitId, int weaponMountId) {
```
Comment (ashdnazg) [inline rts/ExternalAI/SSkirmishAICallbackImpl.cpp]: Shouldn't this check for IsAlliedUnit as well?

**Comment by ashdnazg** [comment]
I think it looks ok, I'm not familiar enough with AI though so I'd like someone else to take a look as well

**Comment by rlcevg** [comment]
https://springrts.com/mantis/view.php?id=4805

**Comment by rlcevg** [comment]
What's the hold-up guys?
My concerns about this PR are due to ZK being very latent to new engines, it may stuck with 99 engine as it already did with 91.

Are there just no devpower to review it? (Sorry if i'm being impatient.)
Or devpower really decided to postpone it until version 100? (I guess I can live with that.)

Tested Weapon with ZK + ai's Cpp, JavaOO.

**Comment by ashdnazg** [comment]
commit currently can't be merged automatically

**Comment by rlcevg** [comment]
Rebased, retested (only Cpp ai)

**Comment by ashdnazg** [comment]
Thanks.

**Pull request merged**